pkg/scheduler/framework/plugins/tainttoleration/taint_toleration.go

```diff
@@ -61,15 +61,18 @@ func (pl *TaintToleration) Filter(ctx context.Context, state *framework.CycleSta
 	if err != nil {
 		return framework.NewStatus(framework.Error, err.Error())
 	}
+	var taintNotTolerated *v1.Taint
 
-	if v1helper.TolerationsTolerateTaintsWithFilter(pod.Spec.Tolerations, taints, func(t *v1.Taint) bool {
+	if taintNotTolerated = v1helper.TaintNotToleratedWithFilter(taints, pod.Spec.Tolerations, func(t *v1.Taint) bool {
 		// PodToleratesNodeTaints is only interested in NoSchedule and NoExecute taints.
 		return t.Effect == v1.TaintEffectNoSchedule || t.Effect == v1.TaintEffectNoExecute
-	}) {
+	}); taintNotTolerated == nil {
 		return nil
 	}
 
-	return framework.NewStatus(framework.UnschedulableAndUnresolvable, ErrReasonNotMatch)
+	errReason := fmt.Sprintf("node(s) had taint {%s: %s}, that the pod didn't tolerate",
```

Comment:
remove the now unused (I assume) ErrReasonNotMatch variable.

Reply:
it's actually used in a generic_scheduler test, (the reason string isn't being compared against though)

Reply:
oops, I think you forgot about this.

Reply:
Uhm... it's wrong that we are using these error messages for no reason if they are not being checked. And at least for the purpose of `TestNodesWherePreemptionMigthHelp`, they don't need to be checked. We should remove those usages and unexport the variables, if possible.

Another good follow up PR :)

Reply:
@starizard any chance you can clean that up?